liiatools/datasets/shared_functions/common.py

```diff
@@ -94,6 +94,34 @@ def save_year_error(input, la_log_dir):
         )
 
 
+def check_year(string):
+    """
+    Check a string to see if it contains a year, if it does, return that year
+    Expected year formats within string:
+        2022
+        14032021
+        2017-18
+        201819
+
+    :param string: String that probably contains a year
+    :return: Year withing the string
+    """
+    match = re.search(r"(20)\d{2}(.*\d)*", string)
+    try:
+        if len(match.group(2)) == 2:
+            year = match.group(1) + match.group(2)
+            return year
+        if len(match.group(2)) == 3:
+            year = match.group(1) + match.group(2)[1:]
+            return year
+        if len(match.group(2)) == 4:
+            year = match.group(2)
+            return year
+    except TypeError:
+        year = match.group(0)
```

Comment:
Have renamed the variable. If there are no regex matches it will throw an AttributeError which is caught just after this function is used
`    try:
        year = common.check_year(filename)
        data["YEAR"] = year
        return data
    except AttributeError:
        common.save_year_error(input, la_log_dir)
        raise Exception(
            f"{filename} was not processed as the filename did not contain a year, "
            f"this error has been sent to the LA to be fixed"
        )`

Reply:
My problem was that in the exception itself, there could be an exception raised from the line I flagged: `year = match.group(0)`  The only reason you'd be in that exception is if match.group caused an error (e.g. there was no match). To call the same function inside the exception itself would potentially break things.  Does that make sense?  In that scenario, the code wouldn't even make it back to the exception you copied above is my concern.